### ISSUE TITLE
Group Rows

### ISSUE THREAD

**Issue by Florent78** [dfinke/ImportExcel#556]
Hello Guys,
I'm wondering how it's possible to group rows.
I know there is a parameter to group rows by date or number (-GroupDateRow).
But I just want to use the simple "Group" function in Excel.
For instance from row 2 to row 6.
If you could give me an example :)

Thanks in advance,

Best Regards,

**Comment by dfinke** [comment]
Could you post a before and after of what'd you'd like to see?

**Comment by Florent78** [comment]
Yes sure.
Here are the before after pictures:

![before](https://user-images.githubusercontent.com/48181002/53697259-06516400-3dd0-11e9-8bd2-a9900d7770de.jpg)

![after](https://user-images.githubusercontent.com/48181002/53697260-0c474500-3dd0-11e9-892c-89908b90be38.jpg)

**Comment by Florent78** [comment]
Do you know if this kind of function is possible with the PowerShell Module ?
Maybe with an Excel Range ?

Thanks

**Comment by pkarunkar** [comment]
I do not think group is available as part of EPPlus.dll . Or atleast the current version is not supporting. 
You may choose to use Excel com Object .

**Comment by dfinke** [comment]
I got this working, let me know if it doesn't work for you.

![image](https://user-images.githubusercontent.com/67258/53844032-726bcd80-3f72-11e9-811a-22fb298d4fa6.png)



```powershell
$xlfile = "$env:TEMP\testOutline.xlsx"
rm $xlfile -ErrorAction SilentlyContinue

$excel = ConvertFrom-Csv @"
Name
Saw
Chain
Hammer
Clamp
Driver1
Driver2
Driver3
"@ | Export-Excel $xlfile -PassThru

$ws=$excel.Sheet1

#for ($i = 1; $i -lt $ws.Dimension.Rows; $i++) { 
for ($i = 2; $i -lt 5; $i++) { 
    $ws.Row($i).OutlineLevel = 1
}

Close-ExcelPackage $excel -Show
```

**Comment by Florent78** [comment]
Awsome,
Thanks a lot, it's working :)

Correct me if I'm wrong, but the best way to understand how it's formated in PowerShell is to:
-Create the Excel file modified manually
-Import the file with the commandelet (Import-Excel)
-And then analyze all the attributes of Rows Cells and Columns to find out the configuration

The method is correct right ?
Or the import-Excel just import the data and not all the properties of the cells, rows etc...

**Comment by ryancys** [comment]
How to make the grouped rows to be collapsed or expanded?

Just like what we did in macro like .ShowDetail = True / False?

Any idea?

**Comment by dfinke** [comment]
```powershell
$xlfile = "$env:TEMP\testOutline.xlsx"
Remove-Item $xlfile -ErrorAction SilentlyContinue

$excel = ConvertFrom-Csv @"
Name
Saw
Chain
Hammer
Clamp
Driver1
Driver2
Driver3
"@ | Export-Excel $xlfile -PassThru

$ws = $excel.Sheet1

for ($i = 2; $i -lt 5; $i++) { 
    $ws.Row($i).OutlineLevel = 1
    $ws.Row($i).Collapsed = $true    
}

for ($i = 6; $i -lt 8; $i++) { 
    $ws.Row($i).OutlineLevel = 2
    $ws.Row($i).Collapsed = $false    
}

Close-ExcelPackage $excel -Show
```

**Comment by NGLocanovic** [comment]
Hi,
I'm glad I found this issue :) Works perfectly.
But I was wondering if one could change the "Collapse Direction" as described in [thespreadsheetguru.com/blog/change-excels-group-outline-direction-settings](https://www.thespreadsheetguru.com/blog/change-excels-group-outline-direction-settings).

There it says:
`ActiveSheet.Outline.SummaryRow = xlAbove`

I tried this, but it does not work:
```Powershell
$Worksheet = $ExcelDoc.'2022 08 22'
$Worksheet.Outline.SummaryRow = "xlAbove"
```

Thanks in advance!
